.github/workflows/tests.yml

```diff
@@ -0,0 +1,23 @@
+name: Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    paths:
+      - '.github/workflows/tests.yml'
+      - 'examples/**'
+      - 'modules/**'
+
+jobs:
+  lint-terraform-files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set Terraform
+        uses: hashicorp/setup-terraform@v3
```

Comment:
should we control which version of Terraform is used here?

Reply:
🤔 Since we're only using this for linting + formatting I think this is ok